### PR TITLE
Set the boot order to director mode

### PR DESCRIPTION
**Pull request by asyedham**

### CODE REVIEW

File: setup_undercloud.yml [hunk 1]
```diff
@@ -45,6 +45,46 @@
...
+
+    - name: Set the boot order to director mode (Dell)
+      shell: |
+        source {{ badfish_venv }}/bin/activate
+        echo {{ item.stdout }}
+        python3 badfish.py -H {{ item.stdout }}  -u quads -p {{ chassis_password }} -i "{{ ansible_user_dir }}/badfish/config/idrac_interfaces.yml" -t director
+      args:
+        chdir: "{{ ansible_user_dir }}/badfish/src/badfish"
+      with_items: "{{ host_list.results }}"
+      when: lab_name == "alias" and undercloud_hostname.split('.')[0].split('-')[3] in alias.machine_types.dell
```
Comment (venkataanil) [inline setup_undercloud.yml]: Please use "vendor" variable https://github.com/redhat-performance/jetpack/blob/master/setup_undercloud.yml#L98

File: setup_undercloud.yml [hunk 2]
```diff
@@ -45,6 +45,46 @@
...
+      args:
+        chdir: "{{ ansible_user_dir }}/badfish/src/badfish"
+      with_items: "{{ host_list.results }}"
+      when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.dell
+
+    - name: Set the boot order to director mode (Dell)
+      shell: |
+        source {{ badfish_venv }}/bin/activate
+        echo {{ item.stdout }}
+        python3 badfish.py -H {{ item.stdout }}  -u quads -p {{ chassis_password }} -i "{{ ansible_user_dir }}/badfish/config/idrac_interfaces.yml" -t director
```
Comment (venkataanil) [inline setup_undercloud.yml]: Sometimes this command may fail, please use "retries" and "delay" like in
https://github.com/redhat-performance/jetpack/blob/master/setup_undercloud.yml#L120

File: setup_undercloud.yml [hunk 3]
```diff
@@ -45,6 +45,46 @@
...
+      args:
+        chdir: "{{ ansible_user_dir }}/badfish/src/badfish"
+      with_items: "{{ host_list.results }}"
+      when: lab_name == "alias" and undercloud_hostname.split('.')[0].split('-')[3] in alias.machine_types.dell
+
     - name: Reboot if OS install needed
       block:
         - name: delete overcloud before re install undercloud
```
Comment (venkataanil) [inline setup_undercloud.yml:92]: Need to check if this "delete overcloud" task [1] is impacting boot order

[1] https://github.com/redhat-performance/jetpack/blob/master/setup_undercloud.yml#L50

File: setup_undercloud.yml (hunk 1 above)
Comment (venkataanil) [reply setup_undercloud.yml]: Do we need to reboot the machine after changing the boot order?

File: setup_undercloud.yml [hunk 4]
```diff
@@ -45,6 +45,46 @@
...
+    - name: Install badfish requirements
+      pip:
+        requirements: "{{ ansible_user_dir }}/badfish/requirements.txt"
+        virtualenv_command: /usr/bin/python3 -m venv
+        virtualenv: "{{ badfish_venv }}"
+
+    - name: list oc_instackenv_content
+      shell: |
+        echo "{{ oc_instackenv_content.nodes[item | int].pm_addr }}"
+      with_sequence: 0-{{ (oc_instackenv_content.nodes|length - 1) }}
```
Comment (venkataanil) [inline setup_undercloud.yml:65]: Why do you want to set the director boot order for undercloud node?

File: setup_undercloud.yml [hunk 5]
```diff
@@ -45,6 +45,46 @@
...
+        python3 badfish.py -H {{ item.stdout }}  -u quads -p {{ chassis_password }} -i "{{ ansible_user_dir }}/badfish/config/idrac_interfaces.yml" -t director
+      args:
+        chdir: "{{ ansible_user_dir }}/badfish/src/badfish"
+      with_items: "{{ host_list.results }}"
+      when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.dell
+
+    - name: Set the boot order to director mode (Dell)
+      shell: |
+        source {{ badfish_venv }}/bin/activate
+        echo {{ item.stdout }}
```
Comment (venkataanil) [inline setup_undercloud.yml]: why is this echo needed?

File: setup_undercloud.yml [hunk 6]
```diff
@@ -45,6 +45,46 @@
...
+        python3 badfish.py -H {{ item.stdout }}  -u quads -p {{ chassis_password }} -i "{{ ansible_user_dir }}/badfish/config/idrac_interfaces.yml" -t director
+      args:
+        chdir: "{{ ansible_user_dir }}/badfish/src/badfish"
+      with_items: "{{ host_list.results }}"
+      when: lab_name == "scale" and undercloud_hostname.split('.')[0].split('-')[3] in scale.machine_types.dell
+
+    - name: Set the boot order to director mode (Dell)
```
Comment (venkataanil) [inline setup_undercloud.yml]: This task is repeated.

File: setup_undercloud.yml (hunk 4 above)
Comment (asyedham) [reply setup_undercloud.yml:65]: oc_instackenv_content does not include undercloud node

File: setup_undercloud.yml (hunk 5 above)
Comment (asyedham) [reply setup_undercloud.yml]: forget to remove it

File: setup_undercloud.yml (hunk 1 above)
Comment (asyedham) [reply setup_undercloud.yml]: not sure, but during introspection the nodes are rebooted

Comment (venkataanil) [reply setup_undercloud.yml]: I would suggest add a power cycle command. I used to do that after setting boot order http://pastebin.test.redhat.com/888239

File: setup_undercloud.yml (hunk 3 above)
Comment (asyedham) [reply setup_undercloud.yml:92]: Done

**Comment by asyedham** [comment]
@venkataanil Have done the requested changes could you please have look

**Pull request merged**